pkg/rke2/rke2_linux.go

```diff
@@ -56,16 +58,188 @@ func initExecutor(clx *cli.Context, cfg Config, dataDir string, disableETCD bool
 		cfg.KubeletPath = "kubelet"
 	}
 
+	var controlPlaneResources podexecutor.ControlPlaneResources
+
+	if cfg.ControlPlaneResourceRequests != "" {
```

Comment:
There's a ton of copy-pasted boilerplate code for extracting the requests and limits. I feel like we could do something with a multidimensional map of component-resource-limit and iterate across those, instead of having it all unrolled here.

What do you think about something like this:
https://play.golang.org/p/A79up20daNq

Reply:
This is super efficient compared to what I built. I'll integrate and test this

Reply:
Integrated your suggestion and optimized it a little bit to be more universal.